services/synchronization/presenters.go

```diff
@@ -16,38 +17,68 @@ type SyncJobRunPresenter struct {
 
 // MarshalJSON returns the JobRun as JSON
 func (p SyncJobRunPresenter) MarshalJSON() ([]byte, error) {
-	type SyncTaskRunPresenter struct {
-		Index  int         `json:"index"`
-		Type   string      `json:"type"`
-		Status string      `json:"status"`
-		Error  null.String `json:"error"`
-	}
-	tasks := []SyncTaskRunPresenter{}
-	for index, task := range p.TaskRuns {
-		tasks = append(tasks, SyncTaskRunPresenter{
-			Index:  index,
-			Type:   string(task.TaskSpec.Type),
-			Status: string(task.Status),
-			Error:  task.Result.ErrorMessage,
-		})
-	}
 	return json.Marshal(&struct {
-		JobID       string                 `json:"jobID"`
-		RunID       string                 `json:"runID"`
+		ID          string                 `json:"id"`
+		JobID       string                 `json:"jobId"`
+		RunID       string                 `json:"runId"`
 		Status      string                 `json:"status"`
 		Error       null.String            `json:"error"`
 		CreatedAt   string                 `json:"createdAt"`
 		Amount      *assets.Link           `json:"amount"`
 		CompletedAt null.Time              `json:"completedAt"`
-		Tasks       []SyncTaskRunPresenter `json:"tasks"`
+		Initiator   syncInitiatorPresenter `json:"initiator"`
+		Tasks       []syncTaskRunPresenter `json:"tasks"`
 	}{
+		ID:          p.ID,
 		RunID:       p.ID,
 		JobID:       p.JobSpecID,
 		Status:      string(p.Status),
 		Error:       p.Result.ErrorMessage,
 		CreatedAt:   utils.ISO8601UTC(p.CreatedAt),
 		Amount:      p.Result.Amount,
 		CompletedAt: p.CompletedAt,
-		Tasks:       tasks,
+		Initiator:   p.initiator(),
+		Tasks:       p.tasks(),
 	})
 }
+
+func (p SyncJobRunPresenter) initiator() syncInitiatorPresenter {
+	var eip *models.EIP55Address
+	if p.RunRequest.Requester != nil {
+		coerced := models.EIP55Address(p.RunRequest.Requester.Hex())
+		eip = &coerced
+	}
+	return syncInitiatorPresenter{
+		Type:      p.Initiator.Type,
+		RequestID: p.RunRequest.RequestID,
+		TxHash:    p.RunRequest.TxHash,
+		Requester: eip,
+	}
+}
+
+func (p SyncJobRunPresenter) tasks() []syncTaskRunPresenter {
+	tasks := []syncTaskRunPresenter{}
+	for index, task := range p.TaskRuns {
+		tasks = append(tasks, syncTaskRunPresenter{
+			Index:  index,
+			Type:   string(task.TaskSpec.Type),
+			Status: string(task.Status),
+			Error:  task.Result.ErrorMessage,
+		})
+	}
+	return tasks
+}
+
+type syncInitiatorPresenter struct {
+	Type      string               `json:"type"`
+	RequestID *null.String         `json:"requestId,omitempty"`
```

Comment:
`*null.String` is this just for efficiency? Does `omitempty` work if `null.String` is null and not `RequestID` is null?

Reply:
`omitempty` does not work if `null.String{Valid: false}` is in the field. We end up with

```
{ requestId: null }
```
instead of 
```
{}
```

Reply:
```
--- FAIL: TestSyncJobRunPresenter_Initiators/web (0.00s)
        presenters_test.go:131:
            	Error Trace:	presenters_test.go:131
            	Error:      	"map[type:web requestId:<nil>]" should have 1 item(s), but has 2
            	Test:       	TestSyncJobRunPresenter_Initiators/web
```